File: app/src/main/java/com/example/fitnessapp/objects/Exercise.java
package com.example.fitnessapp.objects;

import java.util.ArrayList;

public class Exercise {

    private int id, indexExercise;
    private String name, type, note;
    private boolean setsAreVisible;
    private ArrayList<Set> sets;

    public Exercise(int id, String name, String type, String note, ArrayList<Set> sets, boolean setsAreVisible,
                    int indexExercise) {
        this.id = id;
        this.name = name;
        this.type = type;
        this.note = note;
        this.sets = sets;
        this.setsAreVisible = setsAreVisible;
        this.indexExercise = indexExercise;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public ArrayList<Set> getSets() {
        return sets;
    }

    public void setSets(ArrayList<Set> sets) {
        this.sets = sets;
    }

    public boolean getSetsAreVisible() {
        return setsAreVisible;
    }

    public void setSetsAreVisible(boolean setsAreVisible) {
        this.setsAreVisible = setsAreVisible;
    }

    public int getIndexExercise() {
        return indexExercise;
    }

    public void setIndexExercise(int indexExercise) {
        this.indexExercise = indexExercise;
    }
}

File: app/src/main/java/com/example/fitnessapp/objects/Exercise.kt
package com.example.fitnessapp.objects

import android.os.Parcel
import android.os.Parcelable

class Exercise(
    @JvmField var id: Int,
    @JvmField var name: String,
    @JvmField var type: String,
    @JvmField var note: String,
    @JvmField var sets: ArrayList<Set>,
    @JvmField var setsAreVisible: Boolean,
    @JvmField var indexExercise: Int,
    @JvmField var isEditMode: Boolean
) : Parcelable {

    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.readString().toString(),
        parcel.createTypedArrayList(Set.CREATOR) ?: arrayListOf(),
        parcel.readByte() != 0.toByte(),
        parcel.readInt(),
        parcel.readByte() != 0.toByte()
    )

    constructor() : this(-1, "", "", "", arrayListOf(), false, -1, false)

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(name)
        parcel.writeString(type)
        parcel.writeString(note)
        parcel.writeTypedList(sets)
        parcel.writeByte(if (setsAreVisible) 1 else 0)
        parcel.writeInt(indexExercise)
        parcel.writeByte(if (isEditMode) 1 else 0)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Exercise> {
        override fun createFromParcel(parcel: Parcel): Exercise {
            return Exercise(parcel)
        }

        override fun newArray(size: Int): Array<Exercise?> {
            return arrayOfNulls(size)
        }
    }
}
